src/plugins/git/render.js

```diff
@@ -1,24 +1,56 @@
 // @flow
 
+import React from "react";
+import Link from "../../app/Link";
 import * as N from "./nodes";
 import type {Repository} from "./types";
+import {type RepoIdString, stringToRepoId} from "../../core/repoId";
+import type {GitGateway} from "./gitGateway";
 
 export function description(
   address: N.StructuredAddress,
-  repository: Repository
+  repository: Repository,
+  gateway: GitGateway
 ) {
   switch (address.type) {
     case "COMMIT": {
       const hash = address.hash;
       const commit = repository.commits[hash];
       if (commit == null) {
         console.error(`Unable to find data for commit ${hash}`);
-        return hash;
+        return <code>{hash}</code>;
       }
-      const {shortHash, summary} = commit;
-      return `${shortHash}: ${summary}`;
+      const repoIdsForCommit = repository.commitToRepoId[hash];
+      const repoIdStrings: $ReadOnlyArray<RepoIdString> =
+        repoIdsForCommit == null
+          ? []
+          : (Object.keys(repository.commitToRepoId[hash]): any);
```

Comment:
Replace `repository.commitToRepoId[hash]` with `repoIdsForCommit`?

Or replace the whole block with

```js
// This `any`-cast courtesy of facebook/flow#6927.
const repoIdStrings: $ReadOnlyArray<RepoIdString> =
  (Object.keys(repository.commitToRepoId[hash] || {}): any);
```

?